### PR TITLE
Enable `UsePkce` by default for Google and Facebook auth

### PR DESCRIPTION
**Pull request by FranklinWhale**
# Enable `UsePkce` by default for Google and Facebook auth

- [X] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [X] You've included unit or integration tests for your change, where applicable.
- [X] You've included inline docs for your change, where applicable.
- [X] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

1. Set `UsePkce` to `true` in `GoogleOptions` and `FacebookOptions`
2. `BuildChallengeUrl` in `GoogleHandler` now calls `base.BuildChallengeUrl`
3. Rename `AddQueryString` to `SetQueryParam` in `GoogleHandler`
4. Refactor `ChallengeWillTriggerRedirection` in `GoogleTests` to check query dictionary instead of string matching

Contributes to #4684

### CODE REVIEW

**Comment by Tratcher** [comment]
Changing a public API signature is not allowed under our breaking changes guidelines. Try the other approach.

**Review by Tratcher — APPROVED** [review]

File: src/Security/Authentication/Google/src/GoogleHandler.cs [hunk 1]
```diff
@@ -58,27 +59,23 @@ protected override string BuildChallengeUrl(AuthenticationProperties properties,
         // Google Identity Platform Manual:
         // https://developers.google.com/identity/protocols/OAuth2WebServer
 
-        var queryStrings = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-        queryStrings.Add("response_type", "code");
-        queryStrings.Add("client_id", Options.ClientId);
-        queryStrings.Add("redirect_uri", redirectUri);
+        var queryStrings = QueryHelpers.ParseQuery(new Uri(base.BuildChallengeUrl(properties, redirectUri)).Query);
```
[the remark this answers is not in the document]
Comment (FranklinWhale) [reply src/Security/Authentication/Google/src/GoogleHandler.cs:63]: Done!

**Comment by FranklinWhale** [comment]
Added comment and fixed the test.

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**